File: Recursion/Easy15_reverseArray.js
function reverserArray(arr, i=0, j= arr.length -1){
    if(i>=j){
        return arr;
    }

    // swap
    const temp = arr[i]
    arr[i] = arr[j]
    arr[j] = temp

    return reverserArray(arr, ++i, --j)
}


console.log(
    reverserArray([1,2,3,4])
)

File: Recursion/Easy15_reverseArray.py
def reverseArray(arr, i,j):
    if(i >= j):
        return arr

    temp = arr[i]
    arr[i]= arr[j]
    arr[j] = temp
    # print(temp, arr) 
    return reverseArray(arr, i+1, j-1)

arr = [1,2,3,4]
print(
    reverseArray(arr, 0, len(arr)-1)
)